test/integration/resourcemanager/systemcomponentsconfig/systemcomponentsconfig_test.go

```diff
@@ -0,0 +1,246 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package systemcomponentsconfig_test
+
+import (
+	"reflect"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/utils/pointer"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("SystemComponentsConfig tests", func() {
+	var (
+		pod   *corev1.Pod
+		nodes []corev1.Node
+	)
+
+	BeforeEach(func() {
+		pod = &corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-",
+				Namespace:    testNamespace.Name,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "foo",
+						Image: "fooImage",
+					},
+				},
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+		for _, n := range nodes {
+			if !reflect.DeepEqual(n.ObjectMeta, metav1.ObjectMeta{}) {
+				continue
+			}
+
+			node := n
+			node.ObjectMeta = metav1.ObjectMeta{
+				GenerateName: "test-",
+				Labels:       nodeLabels(),
+			}
+
+			Expect(testClient.Create(ctx, &node)).To(Succeed())
+		}
+
+		DeferCleanup(func() {
+			Expect(testClient.DeleteAllOf(ctx, &corev1.Node{}, &client.DeleteAllOfOptions{ListOptions: client.ListOptions{
+				LabelSelector: labels.SelectorFromSet(nodeLabels()),
+			},
+			})).To(Succeed())
```

Comment:
You could simplify this:
```suggestion
				Expect(testClient.DeleteAllOf(ctx, &corev1.Node{}, client.MatchingLabels(nodeLabels()))).To(Succeed())
```